pkg/dag/packages.go

```diff
@@ -0,0 +1,222 @@
+package dag
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"chainguard.dev/melange/pkg/build"
+)
+
+// Configuration represents a configuration along with the file that sourced it
+type Configuration struct {
+	*build.Configuration
+	Path string
+}
+
+type Packages struct {
+	configs  map[string][]*Configuration
```

Comment:
Question: What's the motivation for having a slice of pointers to `Configuration` instead of just `Configuration` itself? What would it mean for a slice element to be `nil`?

Reply:
I'm not quite following yet, maybe it was a poorly phrased question on my part. Why `map[string][]*Configuration` instead of `map[string][]Configuration`?